examples/ring_app/src/examples/ring_app/repl.clj

```diff
@@ -0,0 +1,86 @@
+(ns examples.ring-app.repl
+  (:require
+    [puppetlabs.trapperkeeper.services.metrics.metrics-service
+     :refer [metrics-service metrics-webservice ]]
+    [puppetlabs.trapperkeeper.services.webserver.jetty9-service
+     :refer [jetty9-service]]
+    [puppetlabs.trapperkeeper.services.webrouting.webrouting-service
+     :refer [webrouting-service]]
+    [examples.ring-app.ring-app :refer [count-service]]
+    [puppetlabs.trapperkeeper.core :as tk]
+    [puppetlabs.trapperkeeper.app :as tka]
+    [clojure.tools.namespace.repl :refer (refresh)]))
+
+;; This namespace shows an example of the "reloaded" clojure workflow
+;; ( http://thinkrelevance.com/blog/2013/06/04/clojure-workflow-reloaded )
+;;
+;; It's based on the pattern from Stuart Sierra's `Component` library:
+;; ( https://github.com/stuartsierra/component#reloading )
+;;
+;; You can load this namespace up into a REPL and then run `(go)` to boot
+;; and run the sample application.  Then, you can run `(reset)` at any time
+;; to stop the running app, reload all of the necessary namespaces, and start
+;; a new instance of the app.  This means that you can do iterative development
+;; without having to restart the whole JVM.
+;;
+;; You can also view the context of the application (and all of the
+;; trapperkeeper services) via `(context)` (or pretty-printed with
+;; `print-context`).
+
+(def system nil)
+
+(defn init []
+  (alter-var-root #'system
+                  (fn [_] (tk/build-app
+                           [jetty9-service
+                            webrouting-service
+                            metrics-service
+                            metrics-webservice
+                            count-service]
+                           {:global
```

Comment:
Yeah, that makes sense.  We've been a bit inconsistent about that from project to project but I think I agree that referencing the config from the file is generally better.  At least, it cuts down on the amount of maintenance you might have to do later if the config were otherwise duplicated in both the Clojure code and in HOCON files.  In df5c156, I added code to load the config from the HOCON files instead.  I didn't include all of the flexibility we have in Puppet Server - e.g., around having the ability for a separate `user.clj` file with the ability to override the config from Clojure code.  I could add some of that in later if it seems worthwhile but thought it might be best to just keep this simple for now.